tests/test_pedigree.py

```diff
@@ -174,3 +175,153 @@ def test_pedigree_sanity_checks(self):
 
         with pytest.raises(ValueError):
             ped.get_times(individual=ped.individual)
+
+
+def simulate_pedigree(
+    n_founders=8,
+    n_children_prob=None,
+    n_generations=3,
+    random_sex=True,
+) -> tskit.IndividualTable:
+    def choose_sex(random=True, sex=None):
+        # male = 1, female = 2
+        if random:
+            sex = np.random.binomial(n=1, p=0.5) + 1
+        else:  # non-random: alternate male female to ensure equal balance
+            sex = (sex % 2) + 1
+        return sex
+
+    def choose_n_children(n_children_prob, n_families=1):
+        return np.random.choice(
+            a=len(n_children_prob), size=n_families, p=n_children_prob
+        )
+
+    tb = tskit.IndividualTable()
+    tb.metadata_schema = tskit.MetadataSchema(
+        {
+            "codec": "json",
+            "type": "object",
+            "properties": {
+                "sex": {"type": "integer"},
+            },
+            "required": ["sex"],
+            "additionalProperties": True,
+        }
+    )
+
+    if n_children_prob is None:
+        n_children_prob = [0, 0, 1]
+
+    if random_sex:
+        sex = None
+    else:
+        sex = 1  # male=1, female=2
+
+    # first generation
+    n_males = int(n_founders / 2)
+    curr_gen = [range(n_males), range(n_males, n_founders)]
+    for ind in range(n_founders):
+        tb.add_row(parents=[-1, -1], metadata={"sex": (ind >= n_males) + 1})
+
+    # next generations
+    for gen_idx in range(2, n_generations + 1):
+        next_gen = [[], []]
+        avail_pat = np.random.permutation(curr_gen[0])
+        avail_mat = np.random.permutation(curr_gen[1])
+        n_pairs = min(len(curr_gen[0]), len(curr_gen[1]))
+        if n_pairs == 0 and n_children_prob[0] != 1:
+            raise Exception(
+                f"Not enough people to make children in generation {gen_idx}"
+            )
+        pairs = zip(avail_pat[:n_pairs], avail_mat[:n_pairs])
+        n_children_per_pair = choose_n_children(
```

Comment:
Not sure why you wouldn't do this in the loop? We're not worried about efficiency here, something like
```python
for pair in zip(avail_pat[:n_pairs], avail_mat[:n_pairs]):
   num_children = ...
```